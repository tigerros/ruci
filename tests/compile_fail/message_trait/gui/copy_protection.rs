fn a(_: impl ruci::gui::traits::Message) {}
fn b(m: ruci::CopyProtection) {
    a(m);
}

fn main() {}