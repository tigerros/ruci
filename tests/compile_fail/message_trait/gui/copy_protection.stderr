error[E0277]: the trait bound `ruci::CopyProtection: ruci::gui::traits::Message` is not satisfied
 --> tests/compile_fail/message_trait/gui/copy_protection.rs:3:7
  |
3 |     a(m);
  |     - ^ the trait `ruci::gui::traits::Message` is not implemented for `ruci::CopyProtection`
  |     |
  |     required by a bound introduced by this call
  |
  = help: the following other types implement trait `ruci::gui::traits::Message`:
            &ruci::Debug
            &ruci::Go<'_>
            &ruci::IsReady
            &ruci::PonderHit
            &ruci::Position<'_>
            &ruci::Quit
            &ruci::Register<'_>
            &ruci::SetOption<'_>
          and $N others
note: required by a bound in `a`
 --> tests/compile_fail/message_trait/gui/copy_protection.rs:1:14
  |
1 | fn a(_: impl ruci::gui::traits::Message) {}
  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `a`
